### DISCUSSION TITLE
How to force captcha to show for testing while verifying phone number?

### DISCUSSION THREAD

**Discussion by lucasfronza** [invertase/react-native-firebase#5339] [Q&A] [answered]
Hi 👋 

I've found lots of info about disabling captcha for testing (like using `appVerificationDisabledForTesting`), but I actually want to force it to happen on a real device (for both Android and iOS), so we can manually test if the flow is working as expected.

Is there a way to do it?

Thanks!

**Comment by mikehardy** [comment, marked as answer]
I believe the easiest way is to disable background app refresh or test on an emulator or simulator? Those do reCAPTCHA 100% of the time.

There is an API that we haven't wrapped yet that would expose it, if you wanna do a PR :smile: :pray: ?
https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuthSettings#forceRecaptchaFlowForTesting(boolean)

**Reply by lucasfronza** [reply]
Ah, this sounds interesting! Will try to get something by the weekend 😬

**Reply by mikehardy** [reply]
Fantastic! The Readme in the tests directory is the key document, and it is somewhat easier because I think this is not possible on ios, so it's just one platform? Can follow the example of the code setting the other Auth verify toggle. Cheers

**Reply by lucasfronza** [reply]
Cool!
